database/redis/reply/search_result.go

```diff
@@ -0,0 +1,97 @@
+package reply
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/gomodule/redigo/redis"
+	"github.com/moira-alert/moira"
+	"github.com/moira-alert/moira/database"
+)
+
+type searchHighlightStorageElement struct {
+	Field string `json:"field"`
+	Value string `json:"value"`
+}
+
+type searchResultStorageElement struct {
+	ObjectID   string                          `json:"object_id"`
+	Highlights []searchHighlightStorageElement `json:"highlights"`
+}
+
+func toSearchResultStorageElement(searchResult moira.SearchResult) searchResultStorageElement {
+	result := searchResultStorageElement{
+		ObjectID:   searchResult.ObjectID,
+		Highlights: make([]searchHighlightStorageElement, 0, len(searchResult.Highlights)),
+	}
+	for _, highlight := range searchResult.Highlights {
+		result.Highlights = append(result.Highlights, searchHighlightStorageElement{
+			Field: highlight.Field,
+			Value: highlight.Value,
+		})
+	}
+	return result
+}
+
+// GetSearchResultBytes is a function that takes a search result converts it to a storage structure and marshal it to JSON.
+func GetSearchResultBytes(searchResult moira.SearchResult) ([]byte, error) {
+	storageElement := toSearchResultStorageElement(searchResult)
+	return json.Marshal(storageElement)
+}
+
+func toSearchResult(storageElement searchResultStorageElement) moira.SearchResult {
+	result := moira.SearchResult{
+		ObjectID:   storageElement.ObjectID,
+		Highlights: make([]moira.SearchHighlight, 0, len(storageElement.Highlights)),
+	}
+	for _, highlight := range storageElement.Highlights {
+		result.Highlights = append(result.Highlights, moira.SearchHighlight{
+			Field: highlight.Field,
+			Value: highlight.Value,
+		})
+	}
+	return result
+}
+
+// SearchResult is a function that converts redis reply to SearchResult.
+func SearchResult(rep interface{}, err error) (moira.SearchResult, error) {
```

Comment:
I'm nervous that function name doesn't contain any verbs. SearchResult what?

Reply:
Done in a way another functions named in this package 
https://github.com/moira-alert/moira/blob/a56078419152c5a9b85c2f0f7051442345ff1062/database/redis/reply/trigger.go#L87-L104

https://github.com/moira-alert/moira/blob/a56078419152c5a9b85c2f0f7051442345ff1062/database/redis/reply/notification.go#L13-L50